aspnetcore/security/key-vault-configuration.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Azure Key Vault Configuration Provider | Microsoft Docs
+author: guardrex
+description: How to use the Azure Key Vault Configuration Provider to configure an application using name-value pairs loaded at runtime.
+keywords: ASP.NET Core, configuration, Azure Key Vault
+ms.author: riande
+manager: wpickett
+ms.date: 1/16/2017
+ms.topic: article
+ms.assetid: 0292bdae-b3ed-4637-bd67-19b9bb8b65cb
+ms.prod: aspnet-core
+uid: security/key-vault-configuration
+---
+# Azure Key Vault Configuration Provider
+
+By [Luke Latham](https://github.com/GuardRex) and [Andrew Stanton-Nurse](https://github.com/anurse)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/security/key-vault-configuration/sample)
+
+This document explains how to use the Azure Key Vault Configuration Provider to load application configuration values from Azure Key Vault. [Microsoft Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/) is a cloud-based service that helps you safeguard cryptographic keys and secrets used by applications and services.
+
+## When to use the provider
+Use the provider when your application has access to Azure services and requires encrypted storage of its configuration data. Common scenarios include:
+* Controlling access to sensitive configuration data
+* Monitoring and logging of configuration data use
+* Reducing latency, providing automatic scaling, and providing redundancy of configuration data
```

Comment:
> Are these in general the same reasons that Azure would give for using Key Vault

Yes, that's right. I didn't set out following @anurse's outline assuming that you/he wanted more detail or an alternative explanation to what they provide in their docs.

> "reducing latency" compared to what?

Good question :smile: ... I presume compared to hosting something in Azure but pulling sensitive config data from on-prem or hosting an app/config system on-prem on slower hardware, and perhaps with less bandwidth, than Azure can offer. I lifted that from their doc.

> "providing redundancy"

Presumably, they mean that that the entire Azure Key Vault system rests on Azure infrastructure such that Key Vault databases are mirrored in Azure.

I can go a different direction if you like. Do you want to see an introduction that explains each bullet point in its own paragraph?

The doc link to support these claims is the Azure Key Vault doc at https://azure.microsoft.com/en-us/services/key-vault/, which is where I sourced the content here.